Test/bulk cancel translation process

theresantonie opened the pull request:
### Short description
New test module to test the functionality of the `CancelTranslationProcess` class in `page_bulk_actions.py`.

### Proposed changes
Add this test module to the general testing process and the needed test data to `test_data.json`:
pages: [30, 31] (Region: nurnberg)
pagetranslations: [100,101]

### Side effects
Changed expected results for other tests:
sitemap_nurnberg_de.xml
num_expected_queries for nurnberg_de 78 -> 88

### Resolved issues
Fixes: #2915


__________________________________________________
<!-- Keep this link for the potential reviewer -->
[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

david-venhoff commented on integreat_cms/cms/fixtures/test_data.json:
```diff
@@ -3390,6 +3390,52 @@
+      "depth": 1,
+      "parent": null,
+      "region": 1,
+      "explicitly_archived": true,
```
```suggestion
      "explicitly_archived": false,
```

Technically, this bulk action should not be available for archived pages (I still works for them though ^^). But I think it would be better to test the bulk action only for pages that are not archived, as that is the intended use case

theresantonie replied:
page has been un-archived.

david-venhoff commented on integreat_cms/cms/fixtures/test_data.json:
```diff
@@ -3390,6 +3390,52 @@
+      "depth": 1,
+      "parent": null,
+      "region": 1,
+      "explicitly_archived": true,
```
```suggestion
      "explicitly_archived": false,
```

theresantonie replied:
page has been un-archived.

david-venhoff reviewed (commented):
Thanks, this looks mostly good to me.

Could you also verify that the a message is shown that the page is not in translation if the page was not in translation before, and that a message is shown that the status was successfully cancelled if the page was in translation before?

For reference, you could do something like that: https://github.com/digitalfabrik/integreat-cms/blob/develop/tests/cms/views/feedback/test_region_feedback_actions.py#L55-L59

theresantonie commented:
@david-venhoff 
Thank you for the suggestion regarding the log messages.
I will implement it accordingly.

Regarding the archived pages:
There were some problems with other tests not passing, because the new pages were pulled into those tests and messed with the expected results. 
There are many necessary changes and I haven't found a solution to some of them, yet.

theresantonie commented:
@david-venhoff 
I have added the checks for the log messages.
During my tests, the messages were in German, so I had to add support for English and German log messages.

I also un-archived the pages and moved them to the Nürnberg region to reduce the impact on other tests.
This worked well, but the Editor and Management roles don't have the necessary rights in the Nürnberg region,  
this would have to be added in the test_data.json. (But the side effects here are unclear to me, so far.)

MizukiTemma commented:
@theresantonie 
We can specify the language of error/success messages. See [those lines](https://github.com/digitalfabrik/integreat-cms/blob/a8cc206938c44f5bc9ce2e136e0e3db5102bffbc/tests/mt_api/deepl_api_test.py#L92-L94)

theresantonie commented:
@MizukiTemma 
Thank you for the hint. I was already wondering.

JoeyStk commented:
@theresantonie Thank you a lot for your work thus far. Since this PR hasn't changed in a while, I was wondering if you still  want to finish this PR or if someone else should take over :) Whatever works best for you :)

JoeyStk commented on tests/cms/utils/test_cancel_translation.py:
```diff
@@ -0,0 +1,93 @@
+        )
+
+    kwargs = {"region_slug": REGION_SLUG, "language_slug": LANGUAGE_SLUG}
+    page_ids = IDS_PAGES_IN_TRANSLATION_PROCESS + IDS_PAGES_NOT_IN_TRANSLATION_PROCESS
```
```suggestion
    combined_page_ids = IDS_PAGES_IN_TRANSLATION_PROCESS + IDS_PAGES_NOT_IN_TRANSLATION_PROCESS
```
Super nit: I'm not even sure if this actually makes it better, so feel free to reject this suggestion :) But while reading the test line 79 was the only one that wasn't immediately clear to me. But again, looking up what page_ids was took a few seconds and is definitely justifiable

JoeyStk reviewed (approved):
Thank you very much. I think the test makes sense is good to read :) I only have one super nit :)

jonbulz reviewed (approved):
Comments are just nitpicks, so I'll approve already :)

MizukiTemma commented on tests/cms/utils/test_cancel_translation.py:
```diff
@@ -0,0 +1,95 @@
+        )
+    for page_id in IDS_PAGES_NOT_IN_TRANSLATION_PROCESS:
+        assert (
+            not Page.objects.filter(id=page_id)
```
Uhh 🤔 Our repository contains both the way and  `code_style.sh` accepts both. Maybe we can once discuss about this thema in Weekly 😸
